blocks/extensions/__init__.py

```diff
@@ -92,18 +107,22 @@ class SimpleExtension(TrainingExtension):
         If ``True``, :meth:`do` is invoked before the first epoch.
     after_every_epoch : bool
         If ``True``, :meth:`do` is invoked after every epoch.
-    after_every_iteration : bool
-        If ``True``, :meth:`do` is invoked after every iteration.
+    after_every_batch: bool
```

Comment:
It will not grow too long, because here only rather general and frequent conditions will be added. We could in principle add a parameter `on_callbacks` which value is the list of names of callbacks for which `do` should be called unconditionally, but so far I am not convinced that this is gonna be a better interface. Now it is at least homogeneous: a bunch of parameters which are independent of each other.